normandy/recipes/models.py

```diff
@@ -149,12 +170,16 @@ def compute_implementation_hash(self):
         return hashlib.sha1(self.implementation.encode()).hexdigest()
 
     def save(self, *args, **kwargs):
-        # Save first so the upload is available.
-        super().save(*args, **kwargs)
-
         # Update hash
+        old_hash = self.implementation_hash
         self.implementation_hash = self.compute_implementation_hash()
-        super().save(update_fields=['implementation_hash'])
+
+        # Invalidate signature if content has changed
+        if self.implementation_hash != old_hash:
```

Comment:
maybe we should wait for #141 to land because I added django-dirtyfields which you could use for this